assets/terraform-modules/bare-metal/flatcar-linux/kubernetes/ssh.tf

```
@@ -5,11 +5,14 @@ resource "null_resource" "copy-controller-secrets" {
   # Without depends_on, remote-exec could start and wait for machines before
   # matchbox groups are written, causing a deadlock.
   depends_on = [
-    matchbox_group.install,
-    matchbox_group.controller,
-    matchbox_group.worker,
+    module.controller_profile,
   ]
 
+  # Triggered when the Ignition Config changes (used to recreate a controller)
+  triggers = {
+    clc_config = module.controller[count.index].clc_config
```

If the `.id` does not work, then let's use this:

```suggestion
    clc_config     = module.controller[count.index].clc_config
    kernel_console = join(" ", var.kernel_console)
    kernel_args    = join(" ", var.kernel_args)
```